openlibrary/plugins/openlibrary/js/search.js

```diff
@@ -68,3 +105,55 @@ export function initSearchFacets() {
         less($(this).data('header'), start_facet_count, facet_inc);
     });
 }
+
+/**
+ * A successful response from the partials request
+ *
+ * @typedef {Object} PartialsResponse
+ * @property {string} title - The new title for this page
+ * @property {string} sidebar - HTML string for the facets sidebar
+ * @property {string} [activeFacets] - HTML string for the selected facets
+ */
+/**
+ * Using the given search parameters, makes call for a partially rendered facet affordances
+ * and the new title for the page.
+ *
+ * @param {Object} param
+ * @returns {Promise<PartialsResponse>}
+ *
+ * @throws Error when `/partials` response is not in 200-299 range.
+ */
+function fetchPartials(param) {
+    const data = {
+        param: param,
+        path: location.pathname,
+        query: document.location.search
+    }
+    const dataString = JSON.stringify(data)
+
+    return fetch(`/partials.json?${new URLSearchParams({
```

Comment:
Add lang here if it's specified on the page. You basically need to `new URLSearchParams(location.search).get('lang')`, but handle the case where lang is not defined. Everything else will work down stream, you don't have to handle this parameter on the server-side manually. This is needed so I can test that internationalization is working correctly.

Reply:
Have tried this, and it does not work. Since this applies to all partially rendered HTML, I suggest opening a separate issue for adding this functionality to all partials.
In the meantime, you can change the language with one click on the bottom of the page, or two clicks on the top.

Reply:
Ah ok I thought I got this worked for another partial but it might've been a different flow. Sgtm!